ReactSkia/core_modules/RSkAlertManager.cpp

```diff
@@ -0,0 +1,160 @@
+      alertProps["title"].getString(), alertProps["message"].getString()));
+
+  return 0;
+}
+
+void RSkAlertManager::drawAlertMsg() {
+  std::scoped_lock lock(lockMsgDrawing);
+  if (alertPropsList_.empty() || (ALERT_WINDOW_ACTIVE != alertWindowState_)) {
+    RNS_LOG_INFO("Return From draw Alert Msg without drawing Alert Message!!!");
+    return;
+  }
+  double titleWidth, messageWidth;
+  double drawStartPointX, drawStartPointY;
+  std::shared_ptr<Alert> alertRef = alertPropsList_.back();
+  windowDelegatorCanvas->clear(DARK_THEME_BACKGROUND_COLOR);
+  titleWidth = font_.measureText(
+      alertRef->getTitle().c_str(),
+      strlen(alertRef->getTitle().c_str()),
+      SkTextEncoding::kUTF8);
+
+  if (!alertPropsList_.empty()) {
+    drawStartPointX = (mainWindowSize_.width() - titleWidth) / 2;
+    drawStartPointY = (mainWindowSize_.height() - FONT_SIZE) / 2;
+    WindowDelegator::windowDelegatorCanvas->drawSimpleText(
+        alertRef->getTitle().c_str(),
+        strlen(alertRef->getTitle().c_str()),
+        SkTextEncoding::kUTF8,
+        drawStartPointX,
+        drawStartPointY,
+        font_,
+        paint_);
+
+
+    drawStartPointY = ((mainWindowSize_.height() - FONT_SIZE) / 2) + (FONT_SIZE) + OFFSET_VERTICAL;
+    if (alertRef->getMessage().c_str() != NULL) {
+      WindowDelegator::windowDelegatorCanvas->drawSimpleText(
+        alertRef->getMessage().c_str(),
+        strlen(alertRef->getMessage().c_str()),
+        SkTextEncoding::kUTF8,
+        drawStartPointX,
+        drawStartPointY,
+        font_,
+        paint_);
+    }
+  }
+  displayMsgIndex_ = alertPropsList_.size();
+}
+
+void RSkAlertManager::windowReadyToDrawCB() {
+  alertWindowState_ = ALERT_WINDOW_ACTIVE;
+  drawAlertMsg();
+  commitDrawCall();
+}
+
+void RSkAlertManager::createAlertWindow() {
+  mainWindowSize_ = RnsShell::Window::getMainWindowSize();
+  std::function<void()> createWindowCB =
+      std::bind(&RSkAlertManager::windowReadyToDrawCB, this);
+  std::function<void()> forceFullScreenDrawCB =
+      std::bind(&RSkAlertManager::drawAlertMsg, this);
+  setWindowTittle("Alert Window!!!");
+  createWindow(mainWindowSize_, createWindowCB, forceFullScreenDrawCB);
+  if (subWindowKeyEventId_ == -1) {
+    std::function<void(rnsKey, rnsKeyAction, RnsShell::Window*)> handler =
+        std::bind(
+            &RSkAlertManager::onHWKeyHandler,
+            this,
+            std::placeholders::_1,  // KeyValue
+            std::placeholders::_2,  // eventKeyAction
+            std::placeholders::_3); // Window
+    subWindowKeyEventId_ = NotificationCenter::subWindowCenter().addListener("onHWKeyEvent", handler);
+  }
+  font_.setSize(FONT_SIZE);
+  paint_.setColor(DARK_THEME_FONT_COLOR);
+}
+
+void RSkAlertManager::onHWKeyHandler(
+    rnsKey keyValue,
+    rnsKeyAction eventKeyAction,
+    RnsShell::Window *window) {
+  if ((eventKeyAction != RNS_KEY_Press) || (getWindow() != window)) {
```

Comment:
Window check is a temp workaround. So add a TODO comment here. We should avoid passing entire window and try for ID alone.